command/daemon/command.go

```diff
@@ -60,46 +75,54 @@ func New(config Config) (Command, error) {
 		Run:   newCommand.Execute,
 	}
 
-	newCommand.cobraCommand.PersistentFlags().StringSlice(Flag.Config.Dirs, []string{"."}, "List of config file directories.")
-	newCommand.cobraCommand.PersistentFlags().StringSlice(Flag.Config.Files, []string{"config"}, "List of the config file names. All viper supported extensions can be used.")
+	newCommand.cobraCommand.PersistentFlags().StringSlice(f.Config.Dirs, []string{"."}, "List of config file directories.")
+	newCommand.cobraCommand.PersistentFlags().StringSlice(f.Config.Files, []string{"config"}, "List of the config file names. All viper supported extensions can be used.")
 
-	newCommand.cobraCommand.PersistentFlags().String(Flag.Server.Listen.Address, "http://127.0.0.1:8000", "Address used to make the server listen to.")
-	newCommand.cobraCommand.PersistentFlags().String(Flag.Server.TLS.CaFile, "", "File path of the TLS root CA file, if any.")
-	newCommand.cobraCommand.PersistentFlags().String(Flag.Server.TLS.CrtFile, "", "File path of the TLS public key file, if any.")
-	newCommand.cobraCommand.PersistentFlags().String(Flag.Server.TLS.KeyFile, "", "File path of the TLS private key file, if any.")
+	newCommand.cobraCommand.PersistentFlags().String(f.Server.Listen.Address, "http://127.0.0.1:8000", "Address used to make the server listen to.")
+	newCommand.cobraCommand.PersistentFlags().String(f.Server.TLS.CaFile, "", "File path of the TLS root CA file, if any.")
+	newCommand.cobraCommand.PersistentFlags().String(f.Server.TLS.CrtFile, "", "File path of the TLS public key file, if any.")
+	newCommand.cobraCommand.PersistentFlags().String(f.Server.TLS.KeyFile, "", "File path of the TLS private key file, if any.")
 
 	return newCommand, nil
 }
 
 type command struct {
-	// Internals.
-	cobraCommand  *cobra.Command
+	// Dependencies.
 	logger        logger.Logger
-	serverFactory func() server.Server
+	serverFactory ServerFactory
+
+	// Internals.
+	cobraCommand *cobra.Command
+
+	// Settings.
+	viper *viper.Viper
 }
 
 func (c *command) CobraCommand() *cobra.Command {
 	return c.cobraCommand
 }
 
 func (c *command) Execute(cmd *cobra.Command, args []string) {
-	v := c.serverFactory().Config().Viper
+	// We have to parse the flags given via command line first. Only that way we
+	// are able to use the flag configuration for the location of configuration
+	// directories and files in the next step below.
+	microflag.Parse(c.viper, cmd.Flags())
 
 	// Merge the given command line flags with the given environment variables and
 	// the given config files, if any. The merged flags will be applied to the
 	// given viper.
-	err := microflag.Merge(v, cmd.Flags(), v.GetStringSlice(Flag.Config.Dirs), v.GetStringSlice(Flag.Config.Files))
+	err := microflag.Merge(c.viper, cmd.Flags(), c.viper.GetStringSlice(f.Config.Dirs), c.viper.GetStringSlice(f.Config.Files))
```

Comment:
The two steps of `Parse` and `Merge` make sure that we can use the values of the command line flags for `Config.Dirs` and `Config.Files` to know what config files to use for the merge.